cmd/buck/cli/cli.go

```diff
@@ -77,6 +77,9 @@ func Init(baseCmd *cobra.Command) {
 	archiveCmd.Flags().BoolP("yes", "y", false, "Skips the confirmation prompt if true")
 
 	rolesGrantCmd.Flags().StringP("role", "r", "", "Access role: none, reader, writer, admin")
+
+	linksCmd.Flags().BoolP("json", "", false, "Display URL links as json object")
+	linksCmd.Flags().StringP("format", "", "", "Display URL links in the provided format. Options: [json]")
```

Comment:
The second param is a shorthand flag. For `json` is might be `j`. However, I would just not allow shorthand for these flags. You can do that by using the `Bool`/`String` method instead of `BoolP`/`StringP`.

Reply:
Oh, you're setting up two different flags. We could either do `--json` or `--format=json`. I'd say the latter is more flexible. In which case, just delete the `--json` flag. Then use the value of format to compare with a list of supported formats.

Reply:
I thought if the parameter for the shorthand value is an empty string this functionality is disabled.

> You can do that by using the Bool/String method instead of BoolP/StringP.

I will adapt to these changes.

Reply:
> I'd say the latter is more flexible. In which case, just delete the --json flag.

Why not both flags?

Reply:
> Then use the value of format to compare with a list of supported formats.

is it possible to define validation logic to the flag declaration?

Reply:
> Why not both flags?
Have multiple feels overkill and maybe confusing. Have you seen a similar pattern on other CLI tools?

Reply:
> is it possible to define validation logic to the flag declaration?

The lib being used here is https://github.com/spf13/cobra. As far as I know, you can only use validation on position arguments.

Reply:
> Have multiple feels overkill and maybe confusing. Have you seen a similar pattern on other CLI tools?

yes, that may be overkill. I will remove it.

Reply:
>  As far as I know, you can only use validation on position arguments.

When the pull request [841](https://github.com/spf13/cobra/pull/841) is merged then it may be possible.